executorlib/cache/executor.py

```
@@ -20,10 +28,10 @@
     def __init__(
         self,
         cache_directory: str = "cache",
-        resource_dict: Optional[dict] = None,
+        resource_dict: dict = {},
```

Ruff

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)